compiler/il/OMRNode.cpp

```diff
@@ -5046,10 +5046,16 @@ OMR::Node::hasArrayStride()
    return self()->getOpCode().isArrayLength();
    }
 
+bool
```

Comment:
Can you please clarify why the support for pinning array in node extension is only enabled for data address pointer ? i.e. why not use it for all kinds of nodes ?

Reply:
The limitation was sort of picked up while trying to mimic `hasPinningArrayPointer()`behavior*. But we can remove the limitation by checking extension element size instead:
```diff
- OMR::Node::supportsPinningArrayPointerInNodeExtension()
+ OMR::Node::hasPinningArrayPointerInNodeExtension()
   {
- return self()->isDataAddrPointer();
+   int extensionElemNum = _unionBase._extension.getNumElems();
+   return (extensionElemNum >= 6 && self().getPinningArrayPointer() != NULL)
   }
```

\*`hasPinningArrayPointer()`, unlike the name suggests, it doesn't actually check if the node has a pinning array pointer, It just checks if the opCode supports it. A way to replicate that for node extensions was by checking for some sort of flag on the node, which in this case happened to be `isDataAddrPointer()`.

Reply:
Updated code to support pinning array in node extension for all nodes as suggested 👍

Reply:
Thanks, please run as much internal testing as you are able to, to ensure that that change does not cause failures in downstream projects.

Reply:
Change to support pinning array for node extension for all nodes results in failures that needs investigation, opened an issue https://github.com/eclipse/omr/issues/7379 to track that work to be done later.